analysis.js

```diff
@@ -2,70 +2,166 @@ $("#chart-weeklyDatePicker").datetimepicker({
+  if(compareDate(firstDate,lastDate, start, end)) {
+    $("#chart-select-town").empty();  
+    $("#chart-select-town").append("<option value='{0}'>{0}</option>".format('全區'));
+    $("#chart-select-village").empty();
+    $("#chart-select-village").append("<option value='{0}'>{0}</option>".format('全里'));
+    $('#analysis-name').html('<h3 class="text-center">資料載入中...</h3>');
+    chartFetchWeek(firstDate,lastDate,$("#chart-select-country").val(),$("#chart-select-town").val(),$("#chart-select-village").val());
+  }
+});
 $("#chart-select-country").change(function() {
+  var start = $("#chart-weeklyDatePickerStart").val();
+  var firstDate = moment(start, "YYYY-MM-DD").format("YYYY-MM-DD");
+  var end = $("#chart-weeklyDatePickerEnd").val();
+  var lastDate = moment(end, "YYYY-MM-DD").format("YYYY-MM-DD");
+  $("#chart-select-town").empty();
+  $("#chart-select-town").append("<option value='{0}'>{0}</option>".format('全區'));
+  $("#chart-select-village").empty();
+  $("#chart-select-village").append("<option value='{0}'>{0}</option>".format('全里'));
   $('#analysis-name').html('<h3 class="text-center">資料載入中...</h3>');
-
-  window.fetchWeek($("#chart-weeklyDatePicker").val(), function() {
-    updateTownFormAndTitle();
-  });
+  chartFetchWeek(firstDate,lastDate,$("#chart-select-country").val(),$("#chart-select-town").val(),$("#chart-select-village").val());
 });
 
 $("#chart-select-town").change(function() {
+  var start = $("#chart-weeklyDatePickerStart").val();
+  var firstDate = moment(start, "YYYY-MM-DD").format("YYYY-MM-DD");
+  var end = $("#chart-weeklyDatePickerEnd").val();
+  var lastDate = moment(end, "YYYY-MM-DD").format("YYYY-MM-DD");
+  $("#chart-select-village").empty();
+  $("#chart-select-village").append("<option value='{0}'>{0}</option>".format('全里'));
   $('#analysis-name').html('<h3 class="text-center">資料載入中...</h3>');
-
-  window.fetchWeek($("#chart-weeklyDatePicker").val(), function() {
-    appendChart('#chart', $("#chart-weeklyDatePicker").val());
-    updateAnalysisTitle();
-  });
+  chartFetchWeek(firstDate,lastDate,$("#chart-select-country").val(),$("#chart-select-town").val(),$("#chart-select-village").val());
 });
 
-function updateTownFormAndTitle() {
+function chartFetchWeek(firstDate,lastDate,county,town) {
+  console.log(firstDate+lastDate+county+town);
+  var params;
+  var townTaken = false;
+  if(town == '全區' || town == '無資料') {
+    params = {
+      start : firstDate,
+      end : lastDate,
+      county : county
+    };
+  }
+  else {
+    townTaken = true;
+    params = {
+      start : firstDate,
+      end : lastDate,
+      county : county,
+      town : town
+    };
+  }
+  // console.log(params);
+  $.getJSON(
+    "http://54.91.186.134/api/bucket-record/",
+    params,
+    function(data) {
+      // console.log(town);
+      var lookup = {};
+      var items = data;
+
+      // var result = [];
+      if(!townTaken) {
+        townresult.length = 0;
```

Comment:
may use Array.map here, may be more concise.
ex: 
`
var towns = items.map(function(d){ return d.town})
var townresult = $.unique(towns)
`